### PR TITLE
fix: use required confirmations when tx pending

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Missing confirmation for pending transactions.

## How this PR fixes it

If a transaction is pending an the submitted number of confirmations does not meet the required threshold, the number of confirmations shown is that of the threshold. The confirmations in the transaction stepper is also hidden whilst pending as the confirmations array may not contain the last signer.

## How to test it

Create and immediately execute a transaction. Observe the correct number of signers in the summary and stepper, as well as the confirmees no longer visible in the stepper until the transaction has successfully indexed.

## Screenshots
confirmations

### CODE REVIEW

File: src/components/transactions/TxSigners/index.tsx [hunk 1]
```diff
@@ -108,11 +108,14 @@ export const TxSigners = ({
     return null
   }
 
+  const { confirmations, confirmationsRequired, executor } = detailedExecutionInfo
+
+  // Backend doesn't return all confirmations for immediately executed transactions
+  const numberOfConfirmations =
```
Comment (katspaugh) [inline src/components/transactions/TxSigners/index.tsx:114]: Please avoid prepositions in variable names. `confirmationsCount`

File: src/components/transactions/TxSigners/index.tsx [hunk 2]
```diff
@@ -138,41 +141,41 @@ export const TxSigners = ({
...
-        detailedExecutionInfo.confirmations.map(({ signer }) => getConfirmationStep(signer, signer.value))}
-      {detailedExecutionInfo.confirmations.length > 0 && (
-        <StyledStep $state={StepState.CONFIRMED}>
-          <StepLabel icon={<DotIcon />}>
-            <Link component="button" onClick={toggleHide} fontSize="medium">
-              {hideConfirmations ? 'Show all' : 'Hide all'}
-            </Link>
-          </StepLabel>
-        </StyledStep>
+      {!isPending && (
```
Comment (katspaugh) [inline src/components/transactions/TxSigners/index.tsx:144]: I wouldn't hide them even tho the executioner might not be in the list. It's just how Safe works.

File: src/components/transactions/TxSummary/index.tsx [hunk 3]
```diff
@@ -45,14 +45,17 @@ const TxSummary = ({ item, isGrouped }: TxSummaryProps): ReactElement => {
...
   const nonce = isMultisigExecutionInfo(tx.executionInfo) ? tx.executionInfo.nonce : undefined
-  const submittedConfirmations = isMultisigExecutionInfo(tx.executionInfo)
-    ? tx.executionInfo.confirmationsSubmitted
-    : undefined
   const requiredConfirmations = isMultisigExecutionInfo(tx.executionInfo)
     ? tx.executionInfo.confirmationsRequired
     : undefined
+  const submittedConfirmations = isMultisigExecutionInfo(tx.executionInfo)
+    ? // Backend does not update confirmationsSubmitted for immediately executed transactions
+      isPending && tx.executionInfo.confirmationsSubmitted < tx.executionInfo.confirmationsRequired
```
Comment (katspaugh) [inline src/components/transactions/TxSummary/index.tsx:53]: Just checking: `tx.executionInfo.confirmationsRequired` is the total sigs needed, not the leftover sigs, right?

File: src/components/transactions/TxSigners/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/transactions/TxSigners/index.tsx:144]: Theoretically it's backend lag but in hindsight it's better to show what we have. I'll adjust it.

**Review by usame-algan — APPROVED** [review]
🚀

**Pull request merged**